cli/src/main/java/bisq/cli/CliMain.java

```diff
@@ -152,6 +160,46 @@ public static void run(String[] args) {
                     out.println(btcBalance);
                     return;
                 }
+                case getaddressbalance: {
+                    if (nonOptionArgs.size() < 2)
+                        throw new IllegalArgumentException("no address specified");
+
+                    var request = GetAddressBalanceRequest.newBuilder()
+                            .setAddress(nonOptionArgs.get(1)).build();
+                    var reply = walletService.getAddressBalance(request);
+                    out.println(reply.getAddressBalanceInfo());
+                    return;
+                }
+                case getfundingaddresses: {
+                    var request = GetFundingAddressesRequest.newBuilder().build();
+                    var reply = walletService.getFundingAddresses(request);
+                    out.println(reply.getFundingAddressesInfo());
+                    return;
+                }
+                case createpaymentacct: {
+                    if (nonOptionArgs.size() < 2)
+                        throw new IllegalArgumentException("no account name specified");
+
+                    var accountName = nonOptionArgs.get(1);
+
+                    if (nonOptionArgs.size() < 3)
+                        throw new IllegalArgumentException("no account number specified");
+
+                    var accountNumber = nonOptionArgs.get(2);
+
+                    if (nonOptionArgs.size() < 4)
+                        throw new IllegalArgumentException("no fiat currency specified");
```

Comment:
These size-based checks don't actually know if it was the last argument that was ommitted. We might (or might not) go for more interesting interfaces in the future, but in this case we only know that not all arguments were passed.

Reply:
The original motivation for doing it this way was to keep it as simple as possible because this is/was _a reference implementation only_, to show others how to build their own gRPC CLI.  This isn't the final word on the issue, so let's leave it open.

Reply:
I'm not sure I follow. I'll try to rephrase. My concern here is that this behaviour is incorrect in some cases. Suppose you call `createpaymentacct` with arguments `account name`, `number` (`fiat currency` omitted). It will output `"no fiat currency specified"`, which is correct. But, if you call it with arguments `account name`, `fiat currency` (`account number` omitted), it will output `"no fiat currency specified"`, which is incorrect. Or am I mistaken?

Reply:
I will replace this code that assumes correct param order, with a simple argument count check.   The change will be made in the latest in this long PR chain:  PR [4329](https://github.com/bisq-network/bisq/pull/4329).

Reply:
The requested change is in commit 9691f35 , part of PR [4329](https://github.com/bisq-network/bisq/pull/4329)